src/index.js

```diff
@@ -1,26 +1,53 @@
-export function double ( x ) {
-  return x * 2;
-}
 
-export function doubleXTimes ( x, num ) {
-  let result = x;
+// 1. Create a constructor function for Animal
+// - Why is this a constructor if it doesn't return anything?
+// - The 'new' word uses it to make something...
+// - I don't like that... it's too black-box...
+export const Animal = function() {};
 
-  for ( let i = 1; i <= num; i++ ) {
-    // result = result * 2;
-    result = double( result );
-  }
+// 2. Create a prototype for an Animal,
+// - Well, by creating an Object literal... it inherited a prototype from Object already...
+// - How would I create a prototype without adding anything?
+// - This will happen with the creating of speak() but, just following orders
+Animal.prototype = function() {};
 
-  return result;
+// 3. Create a function on Animal ‘s prototype called speak that returns the animal’s sound.
+Animal.prototype.speak = function(sound = Animal.SOUND) {
+  // default argument no bueno?
+  // should I 'if' an argument instead?
+  return 'generic sound';
 }
 
-export function doubleEach ( arr ) {
-  let result = arr.map( double );
+// 4. Create two new constructors for Reptile and Primate
+export const Reptile = function() {};
+export const Primate = function() {}; // ** Unnecessary, right?
 
-  // const result = arr.map( x => double( x ) );
-  // const result = arr.map( function ( x ) {
-  //   return double( x );
-  // });
+// 5. ...with prototypes that inherit from Animal
+Reptile.prototype = Object.create(Animal.prototype);
+Primate.prototype = Object.create(Animal.prototype); // **
 
-  return result;
-}
+// 6. Overwrite the speak  method for each.
+Reptile.prototype.speak = function(sound = Reptile.SOUND) {
+  return sound;
+};
+Primate.prototype.speak = function(sound = Primate.SOUND) {
+  return sound;
+};
+
+// 6b. Looks like we need some SOUNDS...
+// - I didn't understand what this was at first, but it's in the tests...
+// - so, I have to give each constructor a 'constant' I guess...
+// - "On the contrary... It is in uppercase because that's a common convention for constants defined on the constructor rather than on the prototype."
+// - I still don't see the helpfulness of the all caps but I'm sure it's with reason to avoid some confusion
```

Comment:
The all caps is pure convention, probably a hold-over from an older language. If you look at languages like Java and C, constants are defined in all caps by convention. Even though these are not technically constants (JavaScript has no such feature), we list them in all caps to signify our intent.